core/unit_test/TestWithoutInitializing.hpp

```diff
@@ -97,6 +99,44 @@ TEST(TEST_CATEGORY, resize_realloc_no_alloc) {
   listen_tool_events(Config::DisableAll());
 }
 
+TEST(TEST_CATEGORY, realloc_exec_space) {
+#ifdef KOKKOS_ENABLE_CUDA
+  if (std::is_same<typename TEST_EXECSPACE::memory_space,
+                   Kokkos::CudaUVMSpace>::value)
+    GTEST_SKIP() << "skipping since CudaUVMSpace requires additional fences";
+#endif
+
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableFences());
+  using view_type = Kokkos::View<int*, TEST_EXECSPACE>;
+  view_type outer_view, outer_view2;
+
+  auto success = validate_absence(
+      [&]() {
+        view_type inner_view(Kokkos::view_alloc(TEST_EXECSPACE{}, "bla"), 8);
+        // Avoid testing the destructor
+        outer_view = inner_view;
+        Kokkos::realloc(
+            Kokkos::view_alloc(Kokkos::WithoutInitializing, TEST_EXECSPACE{}),
+            inner_view, 10);
+        outer_view2 = inner_view;
+        Kokkos::realloc(Kokkos::view_alloc(TEST_EXECSPACE{}), inner_view, 10);
+      },
+      [&](BeginFenceEvent event) {
+        // FIXME_CUDA FIXME_HIP FIXME_SYCL FIXME_OPENMPTARGET
```

Comment:
This `FIXME` comment should note what's intended here - as I understand it, fences matching the description strings occur on some/all of those backends, and that they're accepted as not 'wrong' for purposes of this test. There's no clarification of why that should be so, though, or what it would take to fix them.

Reply:
I would remove this together with other similar comments/fences in other places immediately after merging.
Basically, this is just for the fences that we got rid of when allowing allocate to take an execution space.

Reply:
0464d0366280f01e3bcda9c690d082164032679e